pkg/template/apis/template/helpers.go

```diff
@@ -47,16 +48,30 @@ func AddObjectsToTemplate(template *Template, objects []runtime.Object, targetVe
 	return nil
 }
 
-// FilterTemplateInstanceCondition returns a new []TemplateInstanceCondition,
-// ensuring that it does not contain conditions of condType.
-func FilterTemplateInstanceCondition(conditions []TemplateInstanceCondition, condType TemplateInstanceConditionType) []TemplateInstanceCondition {
-	newConditions := make([]TemplateInstanceCondition, 0, len(conditions)+1)
+func (templateInstance *TemplateInstance) HasCondition(typ TemplateInstanceConditionType, status kapi.ConditionStatus) bool {
+	for _, c := range templateInstance.Status.Conditions {
+		if c.Type == typ && c.Status == status {
+			return true
+		}
+	}
+	return false
+}
+
+func (templateInstance *TemplateInstance) SetCondition(condition TemplateInstanceCondition) {
+	condition.LastTransitionTime = metav1.Now()
+
+	for i, c := range templateInstance.Status.Conditions {
+		if c.Type == condition.Type {
+			if c.Message == condition.Message &&
```

Comment:
I think the message and reason can vary even between conditions with the same status and type.